src/Microsoft.DotNet.Interactive/Connection/IndiscriminateProxyKernel.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Interactive.Commands;
+using Microsoft.DotNet.Interactive.Events;
+
+namespace Microsoft.DotNet.Interactive.Connection
+{
+    /// <summary>
+    /// Passes all commands on without any preconceived idea of whether it will be handled.
+    /// </summary>
+    /// <remarks>
+    /// <see cref="ProxyKernel"/> only supports SubmitCode, RequestCompletions, RequestDiagnostics,
```

Comment:
I'm wondering if this separate class is needed. `ProxyKernel` should ideally proxy all command types.

Reply:
It wasn't obvious to me whether anything depended on its current behaviour of deciding not to try to handle commands it didn't have prior knowledge of.

If that selectivity is a feature other users of `ProxyKernel` depend on, would it be better to implement this as a filtering mechanism? E.g., `ProxyKernel` could take a delegate that inspects a command and returns a `bool` indicating whether the command should be sent, at which point we could just use a delegate that always returns `true` to get the "proxy everything" behaviour.

So, do we need that selectivity elsewhere?

(Also, in the back of my mind I wasn't sure if we might end up wanting a proxy to have a clear idea of which commands the remote would be able to handle, in which case this "indiscriminate" proxy might be a temporary feature, eventually replace by something that has some knowledge of which command types the client kernel has handlers for.)

Reply:
> eventually replace by something that has some knowledge of which command types the client kernel has handlers for

I think it makes perfect sense to allow the remote kernel to decide which messages it can receive. So I think the existing proxy kernel should be indiscriminate at this point. The next step would be to add support for a command that performs capability discovery, including which commands, magic commands, and languages are supported by a kernel.

Reply:
OK, I've pushed an update that modifies the existing `ProxyKernel` to work in exactly the same way that this new `IndiscriminateProxyKernel` did, and I've now removed `IndiscriminateProxyKernel`.